src/application/components/applicationFormSubsection.tsx

```diff
@@ -108,205 +108,202 @@ const ApplicationFormSubsectionFields = connect(
-        /*
-         * All usages of field components are created here, so this is a fitting place
-         * for some footnotes about them as a whole.
-         *
-         * The options available for each field allow for a wide variety of display styles
-         * for each field type. This complexity is not easy to map back to sensible data
-         * models for either the UI or for the saved answers, though, so some edge cases
-         * are deliberately not supported and may misbehave if put into use regardless.
-         *
-         * These include:
-         * - Multiple options in a checkbox or radio button group that have an extra
-         *   text input attached. Only one such option should have one; if multiple are
-         *   present, the inputs will all modify the same value.
-         * - Extra inputs on options for dropdown selectors. No input will be shown.
-         * - Options specified for non-option field types such as text fields.
-         *   The options will have no effect.
-         * */
+  },
+)(({
+  formName,
+  section,
+  fieldTypeMapping,
+  identifier,
+  change,
+  sectionApplicantType,
+  getValue,
+}: ApplicationFormSubsectionFieldsProps &
+  ApplicationFormSubsectionFieldsInnerProps) => {
+  const renderField = useCallback(
+    (pathName: string, field: FormField) => {
+      /*
+       * All usages of field components are created here, so this is a fitting place
+       * for some footnotes about them as a whole.
+       *
+       * The options available for each field allow for a wide variety of display styles
+       * for each field type. This complexity is not easy to map back to sensible data
+       * models for either the UI or for the saved answers, though, so some edge cases
+       * are deliberately not supported and may misbehave if put into use regardless.
+       *
+       * These include:
+       * - Multiple options in a checkbox or radio button group that have an extra
+       *   text input attached. Only one such option should have one; if multiple are
+       *   present, the inputs will all modify the same value.
+       * - Extra inputs on options for dropdown selectors. No input will be shown.
+       * - Options specified for non-option field types such as text fields.
+       *   The options will have no effect.
+       * */
 
-        if (!field.enabled) {
-          return null;
-        }
+      if (!field.enabled) {
+        return null;
+      }
 
-        useEffect(() => {
-          // set default value if exists and current value is empty
-          if (
-            field.default_value !== null &&
-            !getValue(`${identifier}.fields.${field.identifier}.value`)
-          ) {
+      useEffect(() => {
+        // set default value if exists and current value is empty
+        if (
+          field.default_value !== null &&
+          !getValue(`${identifier}.fields.${field.identifier}.value`) &&
+          field.identifier !== 'radiobutton'
```

Comment:
What's the rationale for this addition? This isn't actually checking the _type_ of the field, which can be a radiobutton, but its identifier specifically. I'm not aware of any reason that the change shouldn't be triggered on radio buttons in any case.